### PR TITLE
Tsmith/collection geneset

### PR DESCRIPTION
**Pull request by Bento007**
#### Reviewers
**Functional:** @MDunitz 

**Readability:** @tihuan 

---

## Changes
- add geneset to Get collection/{collection_uuid} response.
- modify to_dict to ignore specific attributes and relationships.
- modify to_dict to prevent circular recursion through tables.

## Definition of Done (from ticket)
- add genesets as a field in the collection object and update API tests to ensure the information is included

Updated Collection Object
```
{
  "access_type": "READ",
  "contact_email": "sbadajoz@chanzuckerberg.com",
  "contact_name": "Seve Badajoz",
  "created_at": 1613162996.952171,
  "data_submission_policy_version": "1.0",
  "datasets": [],
  "description": "Hello",
  "id": "4826a9e2-68b4-4017-ad91-ba56605c9396",
  "links":  [],
  "name": "Seve's published collection with DOI",
  "obfuscated_uuid": "",
  "updated_at": 1613162996.952182,
  "visibility": "PUBLIC"
  "gene_sets" :  [
      { 
        "gene_set_name": "somename",
        "gene_set_uuid": "uuid",
        "description": "words",
        "linked_datasets": ["dataset_uuid"]
      },
      { 
        "gene_set_name": "another name",
        "gene_set_uuid": "uuid",
        "description": "words",
        "linked_datasets": ["dataset_uuid"]
      }
]
}
```

### CODE REVIEW

File: backend/corpora/common/corpora_orm.py [hunk 1]
```diff
@@ -396,5 +427,5 @@ class DbGenesetDatasetLink(Base, AuditMixin):
 
     geneset_id = Column(String, ForeignKey("geneset.id"), index=True)
     dataset_id = Column(String, ForeignKey("dataset.id"), index=True)
-    dataset = relationship("DbDataset", back_populates="genesets")
-    geneset = relationship("DbGeneset", back_populates="datasets")
+    dataset = relationship("DbDataset", back_populates="gs")
+    geneset = relationship("DbGeneset", back_populates="gs")
```
Comment (MDunitz) [inline backend/corpora/common/corpora_orm.py:431]:
```suggestion
    geneset = relationship("DbGeneset", back_populates="ds")
```

**Review by tihuan — APPROVED** [review]
LGTM!! 🔥 🔥 🔥 Looks really great! Thanks for the informative PR description too 🙏 Super helpful!

File: backend/corpora/common/corpora_orm.py [hunk 2]
```diff
@@ -37,40 +39,61 @@ class TransformingBase(object):
...
+                    continue
+                # Avoid recursive loop between multiple tables.
+                if relation.target in backref:
+                    continue
+                value = getattr(self, attr)
+                if value is None:
+                    if not remove_none:
+                        result[relation.key] = None
+                elif isinstance(value.__class__, DeclarativeMeta):
+                    result[relation.key] = value.to_dict(backref=backref, remove_none=remove_none)
```
Comment (MDunitz) [inline backend/corpora/common/corpora_orm.py:91]: would we ever want to pass through the `remove_attr`?

**Review by MDunitz — APPROVED** [review]
lgtm! great job handling the recursive relationships

File: backend/corpora/common/corpora_orm.py (hunk 2 above)
Comment (Bento007) [reply backend/corpora/common/corpora_orm.py:91]: There are not uses cases for us that I can think of. It's easy enough to add in later if we want.

**Pull request merged**